### PR TITLE
[P4-2371] [P4-2372] Show cancelled items within filter component

### PR DESCRIPTION
**Pull request by teneightfive**
### What changed

This moves cancelled items into the filter component on collection that use the table layout.

This affects:

- allocation
- single requests

### Why did it change

To build the current layout we need to make two calls to the API to render both active and cancelled results on the page. This has caused some performance issues and with the growing number of results it's not very efficient. Having the filter component means we can move cancelled items to an item in the filter and the results will only be shown when requested.

### Issue tracking

- [P4-2371](https://dsdmoj.atlassian.net/browse/P4-2371)
- [P4-2372](https://dsdmoj.atlassian.net/browse/P4-2372)

## Screenshots

| Before | After |
| ------ | ----- |
| <kbd>localhost_3000_moves_week_2020-08-14_54b75c66-61dd-440e-be3d-b3c58c90be1d_requested_status=pending (1)</kbd> | <kbd>localhost_3000_moves_week_2020-08-14_54b75c66-61dd-440e-be3d-b3c58c90be1d_requested_status=pending</kbd> |
| <kbd>localhost_3000_allocations_week_2020-10-14_outgoing_status=unfilled (1)</kbd> | <kbd>localhost_3000_allocations_week_2020-10-14_outgoing_status=unfilled</kbd> |

#### Automated testing

- [x] Added unit tests to cover changes
- [ ] Added end-to-end tests to cover any journey changes

#### Manual testing

Has been tested in the following browsers:

- [x] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [x] No environment variables were added or changed

### Other considerations

Commit messages with a `fix` or `feat` type are automatically used to generate the [changelog](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/CHANGELOG.md) and
[GitHub release notes](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/releases) during the release task. Please make sure they will read well on their own in a
summary of changes and that the commit body gives a more detailed description of those changes if necessary.

- [x] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics
- [x] Update [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md) with any new instructions or tasks

### CODE REVIEW

**Comment by solidgoldpig** [comment]
What happened to the 'Unfilled allocations' filter?

**Review by Andy-Millar — APPROVED** [review]
Ahhhh now I know why this came up.

Top stuff - looks good

**Comment by Andy-Millar** [comment]
@solidgoldpig Yeap - spotted it before I could get back to you :P

File: app/allocations/middleware/set-results.allocations.js [hunk 1]
```diff
@@ -3,34 +3,24 @@ const allocationService = require('../../../common/services/allocation')
...
-    ).map(response => response.flat())
+    const results = await allocationService.getByDateAndLocation(
+      req.body.allocations
+    )
 
     req.results = {
-      active: activeAllocations,
-      cancelled: cancelledAllocations,
+      active: results,
+      cancelled: [],
```
Comment (solidgoldpig) [inline app/allocations/middleware/set-results.allocations.js:19]: Is there any reason for the cancelled property here any more? (Likewise for the Single Requests)

**Review by solidgoldpig — APPROVED** [review]

File: app/allocations/middleware/set-results.allocations.js (hunk 1 above)
Comment (teneightfive) [reply app/allocations/middleware/set-results.allocations.js:19]: So I believe this is still being used by the "download"...or maybe it's not actually?

But I think I might investigate and clean that up in a separate PR...

Comment (solidgoldpig) [reply app/allocations/middleware/set-results.allocations.js:19]: Pretty sure we're not

https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/f826769d604ab6ff3137fde5a7ebceb9b93f13d6/common/services/move.js#L199

But yeah, separate PR is a good idea. I was just mentioning it…

Comment (teneightfive) [reply app/allocations/middleware/set-results.allocations.js:19]: Yeah good spot. Lemme take a quick look. Might be able to do as a commit on this one as is kinda related...

**Review by jason-sims — APPROVED** [review]
LGTM

File: common/templates/layouts/collection.njk [hunk 2]
```diff
@@ -72,7 +72,7 @@
       {% block headerLinks %}
         <div class="govuk-grid-column-one-third">
           <ul class="govuk-list govuk-!-font-size-16 app-!-float-right app-print--hide">
-            {% if canAccess('moves:download') %}
+            {% if canAccess('moves:download') and context !== "allocations" %}
```
Comment (solidgoldpig) [inline common/templates/layouts/collection.njk:75]: maybe this is something that could be handled by the prohibitions stuff. If we can prohibit by locationType, why not be `context`?

**Review by solidgoldpig — APPROVED** [review]

File: common/templates/layouts/collection.njk (hunk 2 above)
Comment (teneightfive) [reply common/templates/layouts/collection.njk:75]: I looked at doing that but it's a bit of a rabbit hole at the moment. The app doesn't actually currently support downloading allocations either so I think we need to address this separately.

**Pull request merged**